clang/include/clang/Basic/Builtins.td

```diff
@@ -4518,6 +4518,12 @@ def HLSLCreateHandle : LangBuiltin<"HLSL_LANG"> {
   let Prototype = "void*(unsigned char)";
 }
 
+def HLSLDotProduct : LangBuiltin<"HLSL_LANG"> {
+  let Spellings = ["__builtin_hlsl_dot"];
+  let Attributes = [NoThrow, Const, CustomTypeChecking];
+  let Prototype = "void(...)";
```

Comment:
The return type should not be void.

Reply:
Thats a place holder `clang\lib\Sema\SemaExpr.cpp` has these `hasCustomTypechecking` checks  that call `CheckBuiltinFunctionCall` in `clang\lib\Sema\SemaChecking.cpp` I want to do custom type checking but that piece was dragging the pr and I wanted to get some feedback. Since it isn't ready yet so I pulled it from the pr and disabled code gen for the bad cases as a stop gap